spec/active_record/events_spec.rb

```diff
@@ -41,6 +41,19 @@
     it 'defines an inverse scope' do
       expect(Task.not_completed).to include(task)
     end
+
+    it 'allows overriding methods defined by the gem' do
+      task.class.class_eval do
```

Comment:
This is neat, but it permanently adds the method to the `Task` model (not only for this example but globally). To avoid confusion, I would define the method explicitly in the model class inside the dummy application (`spec/dummy/app/models/task.rb`), which would have the same effect. It could be something like:

```ruby
class Task < ActiveRecord::Base
  has_event :complete

  def complete!
    super
    logger.info("Task #{id} has been completed")
  end
end
```